lib/avro/builder/field.rb

```diff
@@ -30,8 +30,10 @@ def initialize(name:, type_name:, record:, builder:, internal: {}, options: {},
           send(key, value) if has_dsl_attribute?(key)
         end
 
-        @type = builder.lookup(type_name, required: false) ||
-          build_type(type_name, field: self, internal: internal, options: options)
+        @type = build_type(type_name,
```

Comment:
I think I'm missing something here. I thought the intent was to looking an existing type or create a new type. Do we not want that behavior?